pydantic/_internal/_fields.py

```diff
@@ -53,6 +56,53 @@ def get_type_hints_infer_globalns(
     return get_type_hints(obj, globalns=globalns, localns=localns, include_extras=include_extras)
 
 
+def get_typed_dict_type_hints(
+    typed_dict_cls: Any,
+    types_namespace: dict[str, Any] | None = None,
+) -> dict[str, Any]:
+    """Collect the type hints of a TypedDict.
+
+    This takes into account any TypeVar transformations when using generic
+    TypedDict.
+
+    Args:
+        typed_dict_cls: a TypedDict subclass
+        types_namespace: Optional extra namespace to look for types in.
+
+    Returns:
+        the final type hints.
+    """
+    hints: dict[str, Any] = {}
+    typevars_map: dict[type, dict[Any, Any]] = {}
+    previous_base_cls: type | None = None
```

Comment:
Given the difference between `base` and `base_cls` below, I think this name is better:

```suggestion
    previous_base: type | None = None
```

However, I also half wonder if looking just at the previous base type is the reason why the test starts failing for me once I add another layer as per other comment.